src/metaxy/metadata_store/lancedb.py

```diff
@@ -0,0 +1,333 @@
+            Tables are created automatically when writing metadata.
+
+        Example:
+            ```py
+            # Local directory
+            store = LanceDBMetadataStore("./metadata")
+
+            # With custom hash algorithm
+            store = LanceDBMetadataStore(
+                "./metadata",
+                hash_algorithm=HashAlgorithm.SHA256
+            )
+
+            # S3 backend (requires AWS credentials)
+            prod = LanceDBMetadataStore("s3://prod-bucket/metadata")
+
+            # LanceDB Cloud (requires LANCEDB_API_KEY environment variable)
+            cloud = LanceDBMetadataStore("db://my-database")
+
+            # Local with fallback to production
+            dev = LanceDBMetadataStore("./dev", fallback_stores=[prod])
+            ```
+        """
+        # Store as string to handle both local paths and remote URIs
+        self.database_path = str(database_path)
+        self._conn: Any | None = None
+        super().__init__(fallback_stores=fallback_stores, **kwargs)
+
+    def _get_default_hash_algorithm(self) -> HashAlgorithm:
+        """Default hash algorithm."""
+        return HashAlgorithm.XXHASH64
+
+    def _supports_native_components(self) -> bool:
+        """LanceDB store uses Polars components."""
+        return False
+
+    def native_implementation(self) -> nw.Implementation:
+        """LanceDB operations run via Polars/Narwhals."""
+        return nw.Implementation.POLARS
+
+    @contextmanager
+    def _create_provenance_tracker(self, plan):
+        """Create a Polars provenance tracker for LanceDB."""
+        from metaxy.provenance.polars import PolarsProvenanceTracker
+
+        tracker = PolarsProvenanceTracker(plan=plan)
+        try:
+            yield tracker
+        finally:
+            # No cleanup needed for Polars tracker
+            pass
+
+    def _create_native_components(self) -> tuple[Any, Any, Any]:
+        """Not supported - LanceDB relies on Polars fallback components."""
+        raise NotImplementedError(
+            "LanceDBMetadataStore does not support native field provenance calculations"
+        )
+
+    def open(self) -> None:
+        """Open LanceDB connection.
+
+        For local filesystem paths, creates the directory if it doesn't exist.
+        For remote URIs (S3, LanceDB Cloud, etc.), connects directly.
+        Tables are created on-demand when features are first written.
+
+        Raises:
+            ImportError: If lancedb package is not installed
+            ConnectionError: If remote connection fails (e.g., invalid credentials)
+        """
+        import lancedb
+
+        # Only create directory for local filesystem paths
+        # Remote URIs (s3://, db://, http://, https://, gs://, etc.) are handled by LanceDB
+        if not self._is_remote_uri(self.database_path):
+            Path(self.database_path).mkdir(parents=True, exist_ok=True)
+
+        self._conn = lancedb.connect(self.database_path)
```

Comment:
Would we want to enable async mode? 

https://lancedb.com/docs/integrations/platforms/polars_arrow/